ecosystem/indexer-grpc/indexer-grpc-integration-test-transaction-generator/src/move_executor.rs

```diff
@@ -0,0 +1,305 @@
+        Ok(())
+    }
+
+    /// Load the config from .aptos/config.toml
+    async fn load_config(&self, move_module_folder: PathBuf) -> anyhow::Result<Profile> {
+        let config_path = move_module_folder.join(".aptos/config.yaml");
+        let config_content = tokio::fs::read_to_string(&config_path).await?;
+        let config: AptosCliConfig = serde_yaml::from_str(&config_content)
+            .context(format!("Failed to parse config file at {:?}.", config_path))?;
+        Ok(config.profiles.default)
+    }
+
+    /// Fund the account with the faucet.
+    async fn fund_account(&self, amount: u64) -> anyhow::Result<()> {
+        let mut fund_account_cmd = tokio::process::Command::new(APTOS_CLI_BINARY_NAME);
+        fund_account_cmd
+            .arg("account")
+            .arg("fund-with-faucet")
+            .arg("--amount")
+            .arg(amount.to_string())
+            .arg("--profile")
+            .arg(DEFAULT_PROFILE_NAME)
+            .arg("--faucet-url")
+            .arg(LOCAL_FAUCET_URL);
+        let output = fund_account_cmd.output().await?;
+        if !output.status.success() {
+            return Err(anyhow::anyhow!("Failed to fund account: {:?}", output));
+        }
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[tokio::test]
+    async fn test_config_parsing() {
+        let raw_yaml_content = r#"---
+            profiles:
+                default:
+                    amount_to_fund: 100000000
+                    is_setup: true
+                testing:
+                    private_key: "0x123"
+                    public_key: "0x123"
+                    account: 456
+                    rest_url: "http://localhost:8080"
+                    faucet_url: "http://localhost:8081"
+                    # Testing Related
+                    amount_to_fund: 100000001
+        "#;
+        let config: AptosCliConfig = serde_yaml::from_str(raw_yaml_content).unwrap();
+        assert_eq!(config.profiles.default.amount_to_fund, 100000000);
+        assert!(config.profiles.default.is_setup);
+    }
+
+    #[tokio::test]
+    async fn test_aptos_cli_output_parsing() {
+        let raw_json_content = r#"{
+            "Result": {
+                "transaction_hash": "0x568e3024d8eca50629ccb01937275506f8dfc848d3dfb1f4701690cc9b412b33",
+                "gas_used": 3,
+                "gas_unit_price": 100,
+                "sender": "49cbfc8d9fc24297df19a47cc05c2fb194d1bebf15b85124bf8f346ffde8eccc",
+                "sequence_number": 0,
+                "success": true,
+                "timestamp_us": 1718143556686324,
+                "version": 79373359,
+                "vm_status": "Executed successfully"
+            }
+        }"#;
+        let output: AptosCliOutput = serde_json::from_str(raw_json_content).unwrap();
+        assert_eq!(output.result.version, 79373359);
+    }
+
+    #[tokio::test]
+    async fn test_move_config_parsing() {
+        let raw_toml_content = r#"
```

Comment:
How come we need to test this, parsing Move.toml seems like something we should already have tests for elsewhere.